### ISSUE TITLE
Tests failing on master

### ISSUE THREAD

**Issue by gsnedders** [w3c/web-platform-tests#10355]
https://github.com/w3c/web-platform-tests/pull/10335 is another case of some change having broken an unrun test and that successfully landing on master.

Should we enable a [Travis cron job](https://docs.travis-ci.com/user/cron-jobs/) and once a day run all the jobs (probably also running lint on everything, too)?

**Comment by Hexcles** [comment]
On a side note, perhaps we can stop skipping some of the jobs?

e.g. https://travis-ci.org/w3c/web-platform-tests/builds/362774898
`ci_wpt.sh` only takes 12min. Is it forbiddenly slow to always run?

**Comment by foolip** [comment]
I've put this in the list for [Resolve known important infra issues](https://docs.google.com/document/d/1uh7A24pT6P70Lb9CeFCKga3RrrdzLK7tSAR7o8C7No0/edit#heading=h.3sszqdrf0hk7).

**Comment by foolip** [comment]
@gsnedders, this might have been fixed by @Hexcles already, can you take a look at the original problem?

**Comment by gsnedders** [comment]
@Hexcles fixed the original problem, probably well enough?

I think my intention here was some discussion about more generally fixing the issue where we skip tests as irrelevant but silently break them.

**Comment by jgraham** [comment]
So I don't know that there's an easy way of running the right set of tests without either running everything, which is hugely inefficient particuarly given how capacity-limited travis is, or having some map of changes to tests. We currently do the latter manually and it's possible to imagine doing it automatically with sufficient coverage data (you compute the coverage per test(job) and then reverse that map to get the set of tests that might be affected for a given change.

If we move away from Travis the capacity issues may be reduced, which will help.

**Comment by gsnedders** [comment]
The sensible thing is to probably run all the unit tests on every commit on master, given we want to know ASAP that they're broken so we can revert. I mean, arguably even we want to run the unit tests on every PR. That said, we definitely don't have the capacity to do either reasonably.

Running all the unit tests once a day on master is potentially a good compromise (how often do unit tests get run on average at the moment? probably less often that that, which means our time-to-detection can be poor), I think?

**Comment by foolip** [comment]
Looking at https://travis-ci.org/web-platform-tests/wpt/builds/414118543 it seems like running the unit tests only takes 3 minutes. Isn't that acceptable to do on every PR?

**Comment by gsnedders** [comment]
That took the failure shortcut; I think https://travis-ci.org/web-platform-tests/wpt/jobs/414181137 is all, double the time at six minutes.

**Comment by foolip** [comment]
Included in [Ecosystem Infra 2018 Q4 OKRs](https://docs.google.com/document/d/1UKuPaQm2YGHYEfbrrYBHoCUGAVhyQZEp8nE09qT7vYg/edit#heading=h.jwokykbtd9cq)

**Comment by mdittmer** [comment]
_Ping from your friendly neighbourhood ecosystem infra rotation_

Any updates on this, @gsnedders?

**Comment by gsnedders** [comment]
Not been worked on.

**Comment by foolip** [comment]
@gsnedders any chance of getting this fixed before the end of the quarter?

**Comment by gsnedders** [comment]
I don't think we ever got a clear idea as to what to do here. I remain in favour of running unit tests on master, which we currently don't, and @jgraham is against doing so as far as I'm aware.

**Comment by foolip** [comment]
@jgraham don't you want to do that because of capacity? How about if we just do it once a day?

**Comment by gsnedders** [comment]
On the whole I think we should close this as wontfix? Probably not worth addressing, given how few tests we reference in the unit tests?

**Comment by stephenmcgruer** [comment]
I strongly disagree about closing this wontfix. Unittests that aren't run (regularly) are just false peace-of-mind - we think we have test coverage, but the actual truth is that we don't. They should either be run, or deleted.

That said, it's not clear to me from this issue what unit tests aren't run. Do we have a list?


In the case of https://github.com/web-platform-tests/wpt/pull/10335 - that seems like a bad test to me. A unittest should not rely on the actual production system (in this case a particular file), it should instead mock out the system as necessary. If the argument is that testing `tests-affected` surmounts to just testing `git` (I've not checking if there's more logic to it than just `git diff`), then I believe the correct action is to trust git's own testing and remove the test from our infrastructure. Alternatively, we could make the test just a smoke-test, i.e. by just checking a commit that we know should have *some* diff and make sure there is *some* files in the output.

**Comment by gsnedders** [comment]
I think it is just the case of #10335?

**Comment by Hexcles** [comment]
> That said, it's not clear to me from this issue what unit tests aren't run. Do we have a list?

We are not running any unit tests per se on master. We only run them on PRs when `infra/` is changed. You can pick a random "infra" PR to see the list of unit tests ([example](https://community-tc.services.mozilla.com/tasks/groups/HjT9pqZjQoWoGQXjkDEy0w)).

#10335 was the only case we've seen that this has ever become a problem AFAIK.

**Comment by stephenmcgruer** [comment]
So I think the main action item here then is to address the test in #10335, fixing it up to not rely on the actual repository status. Then as a secondary action item, it would be nice to run the unit tests on master (if we're sure we'd notice failures), but its not critical if we think that the current manually-specified dependency calculation (aka changes `infra/`) is sufficient.
